x-pack/plugins/security/server/routes/authentication/common.ts

```diff
@@ -71,4 +76,53 @@ export function defineCommonRoutes({ router, authc, basePath, logger }: RouteDef
       })
     );
   }
+
+  function getLoginAttemptForProviderType(providerType: string, next: string) {
+    if (providerType === SAMLAuthenticationProvider.type) {
+      return { type: SAMLLogin.LoginInitiatedByUser, redirectURL: next };
+    }
+
+    if (providerType === OIDCAuthenticationProvider.type) {
+      return { type: OIDCLogin.LoginInitiatedByUser, redirectURL: next };
+    }
+
+    return undefined;
+  }
+
+  router.get(
```

Comment:
__note:__ we need to properly handle redirects that's why I'm using `GET` here.

Reply:
Redirects from other web-servers using a 301/302?

Reply:
I meant that this endpoint responds with 302 and we won't be able to easily handle this on the client side if use HTTP client do that that `POST`. 

We can alternatively use hidden `<form />` to do the `POST` here and mark route as `requireXsrf: false` and somehow preserve current hash fragment (I'm not sure if browser preserves it if we get redirect on form submit like it does for normal redirects), but I'm not sure how I feel about that added complexity yet...

What do you think? Am I missing something here?

Reply:
I think it's worth seeing how the hidden `<form />` works or using a XHR and manually doing the redirect. If either of these aren't super ugly, they'd add a bit of additional protection.